File: python/courtlistener/api/recap_query.py
"""
RECAP Query API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class RecapQueryAPI(BaseAPI):
    """API for querying RECAP database."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "recap-query/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List RECAP query operations.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of RECAP query operations
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, query_id: int) -> Dict[str, Any]:
        """
        Get a specific RECAP query operation by ID.
        
        Args:
            query_id: RECAP query operation ID
        
        Returns:
            RECAP query operation data
        """
        return self.client.get(f"{self.endpoint}{query_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated RECAP query operations.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for RECAP query operations
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/RecapQuery.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * RecapQuery API endpoint
 */
class RecapQuery extends BaseApi
{
    protected string $endpoint = 'recap-query/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List recap query with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listRecapQuery(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific recap query by ID
     *
     * @param int|string $id RecapQuery ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getRecapQuery(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search recap query
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchRecapQuery(array $params = []) {
        return $this->search($params);
    }
}
